### PR TITLE
[API Server] Provider service

### PR DESCRIPTION
**Pull request by aborilov**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apiserver/api/v1/provider.proto [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+message AccountMetadata {
+  string displayName = 1;
+  string description = 2;
+}
+
+service ProviderService {
+  rpc List(ProviderListRequest) returns (ProviderList) {
+    option (google.api.http) = {
+      get : "/v1/tenants/{tenant}/providers"
```
Comment (nmiculinic) [inline pkg/apiserver/api/v1/provider.proto:58]:
```suggestion
      get : "/v1/providers"
```

File: pkg/apiserver/api/v1/provider.proto [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    };
+  };
+  rpc Get(ProviderRequest) returns (Provider) {
+    option (google.api.http) = {
+      get : "/v1/tenants/{tenant}/providers/{name}"
```
Comment (nmiculinic) [inline pkg/apiserver/api/v1/provider.proto:63]:
```suggestion
      get : "/v1/providers/{name}"
```

File: pkg/apiserver/api/v1/provider.proto (hunk 1 above)
Comment (jiachengxu) [reply pkg/apiserver/api/v1/provider.proto:58]: nope, this is the endpoint for listing provider objects in the account namespace, so it should be:
"/v1/accounts/{account}/providers"

Comment (jiachengxu) [reply pkg/apiserver/api/v1/provider.proto:58]: and also for the following "get"

**Pull request merged**